### PR TITLE
Navx edits

### PR DESCRIPTION
**Pull request by ishanvermani**

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/NAVX.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    private static DigitalInput distanceSensorFar;
+    private static boolean shortDistance;
+    private static boolean longDistance;
+
+    private static int thresholdDistance = 800;
+
+
+
+    private static String color;
+
```
Comment (ishanvermani) [inline src/main/java/frc/robot/subsystems/NAVX.java:31]: added every sensor but the NavX lmao

Comment (ishanvermani) [reply src/main/java/frc/robot/subsystems/NAVX.java:31]: See how to add NavX as well as common methods here: https://www.kauailabs.com/public_files/navx-mxp/apidocs/java/com/kauailabs/navx/frc/AHRS.html

File: src/main/java/frc/robot/subsystems/NAVX.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+public class NAVX extends SubsystemBase {
+
+    private AHRS navx;
+    private AnalogInput irSensor1;
+    private LiDAR lidar;
+
+    private RobotContainer container; // This classes reference to the RobotContainer
+
+    public NAVX(RobotContainer container) {
+        this.container = container;
```
Comment (ishanvermani) [inline src/main/java/frc/robot/subsystems/NAVX.java:31]: You don't use container anywhere - remove argument for RobotContainer as well as private variable.

File: src/main/java/frc/robot/subsystems/NAVX.java [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        }
+    
+    }
+
+    @Override
+    public void periodic() {
+    }
+
+    // We could just use 4096 for this, but this method will ensure accuracy if one changes
+
```
Comment (ishanvermani) [inline src/main/java/frc/robot/subsystems/NAVX.java:90]: Put values for X, Y, Z, and lidar distance on smart dashboard.

File: src/main/java/frc/robot/RobotContainer.java [hunk 4]
```diff
@@ -27,12 +28,18 @@ public class RobotContainer {
...
+
 	/**
 	 * The container for the robot. Contains subsystems, OI devices, and commands.
 	 */
 	public RobotContainer() {
 		// Configure the trigger bindings
 		configureBindings();
+
+		//initialized?
+		navx = new NAVX(null);
```
Comment (ishanvermani) [inline src/main/java/frc/robot/RobotContainer.java:42]: see comments about Arguments in NavX Class - there should be nothing passed here. NavX can actually be initialized like the drivetrain class in line 29

**Comment by ishanvermani** [comment]
3 small changes, and then good to merge @Polina-Afa @DevenVElani

File: src/main/java/frc/robot/subsystems/NAVX.java (hunk 3 above)
Comment (ishanvermani) [reply src/main/java/frc/robot/subsystems/NAVX.java:90]: `this.getGyroX()`

**Review by turtle77777777 — APPROVED** [review]

**Pull request merged**